extensions/markdown/syntaxes/markdown.tmLanguage

```diff
@@ -536,25 +540,37 @@
 					<!-- seperator    [ ]{0,3}([-*_][ ]{0,2}\2){2,}[ \t]*$\n? -->
 					<!-- list         [ ]{0,3}[*+-]([ ]{1,3}|\t) -->
 					<!-- both are folded together in the expression below -->
-					<string>(^|\G)(?!\s*$|#|[ ]{0,3}((([*_][ ]{0,2}\2){2,}[ \t]*$\n?)|([*+-]([ ]{1,3}|\t)))|\s*\[.+?\]:|&gt;)</string>
-				</dict>
-				<key>raw_block</key>
-				<dict>
-					<key>begin</key>
-					<string>(^|\G)([ ]{4}|\t)</string>
-					<key>name</key>
-					<string>markup.raw.block.markdown</string>
-					<key>while</key>
-					<string>(^|\G)([ ]{4}|\t)</string>
+					<string>(^|\G)(?!\s*$|#|[`~]{0,3}|[ ]{0,3}((([*_][ ]{0,2}\2){2,}[ \t]*$\n?)|([*+-]([ ]{1,3}|\t)))|\s*\[.+?\]:|&gt;)</string>
 				</dict>
 				<key>fenced_code_block_css</key>
 				<dict>
 					<key>begin</key>
-					<string>(^|\G)\s*(([`~]){3,})\s*(css|css.erb)(\s+.*)?$</string>
+					<string>(^|\G)\s*([`~]{3,})\s*(css|css.erb)\n</string>
 					<key>name</key>
 					<string>markup.fenced_code.block.markdown</string>
-					<key>while</key>
-					<string>(^|\G)(?!\s*\2\3*\s*$)</string>
+					<key>end</key>
+					<string>(^|\G)\s*([`~]{3,})\n</string>
```

Comment:
We have to use a while clause here. This prevents broken language grammars from leaking outside of the fenced block. Switching to while from end fixed a large number of syntax highlighting issues.